Client/Helpers/FxALoginHelper.swift

```diff
@@ -202,8 +203,13 @@ class FxALoginHelper {
         if let profile = self.profile, let account = account {
             profile.setAccount(account)
             // account.advance is idempotent.
-            if  let account = profile.getAccount(), accountVerified! {
-                account.advance()
+            if  let account = profile.getAccount() {
```

Comment:
Yes! It turns out that `accountVerified` is checked by `account.advance()`. Additionally, the registration of the device (including push) only happens on `advance()`. 

Thus, we can't push on verification unless we pre-verify `advance`.